test/new-e2e/tests/apm/docker_test.go

```diff
@@ -23,137 +22,88 @@ import (
 
 type DockerFakeintakeSuite struct {
 	e2e.BaseSuite[environments.DockerHost]
+	transport transport
 }
 
-func TestDockerFakeintakeSuite(t *testing.T) {
-	devModeEnv, _ := os.LookupEnv("E2E_DEVMODE")
-	options := []e2e.SuiteOption{
-		e2e.WithProvisioner(awsdocker.Provisioner(awsdocker.WithAgentOptions(
-			dockeragentparams.WithAgentServiceEnvVariable("DD_APM_RECEIVER_SOCKET", pulumi.String("/var/run/datadog/apm.socket")), // Enable the UDS receiver in the trace-agent
-			dockeragentparams.WithAgentServiceEnvVariable("STATSD_URL", pulumi.String("unix:///var/run/datadog/dsd.socket")),      // Optional: UDS is more reliable for statsd metrics
-		))),
+func dockerSuiteOpts(tr transport, opts ...awsdocker.ProvisionerOption) []e2e.SuiteOption {
+	if !flag.Parsed() {
+		flag.Parse()
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
No need to parse flags in tests, `go test` does it out of the box, except for `TestMain`